homework/shared_ptr/WeakPointer.hpp

```diff
@@ -112,6 +112,7 @@ WeakPointer<T>& WeakPointer<T>::operator=(const WeakPointer<T>& anotherPtr) {
         ptr_ = anotherPtr.ptr_;
         refCounter_ = anotherPtr.refCounter_;
         refCounter_->increaseWeak();
+        checkControlBlock();
```

Comment:
do it before you reassign resources. It should work for the old resource.
```suggestion
        checkControlBlock();
        ptr_ = anotherPtr.ptr_;
        refCounter_ = anotherPtr.refCounter_;
        refCounter_->increaseWeak();
```